.github/workflows/v2-deploy-demos-swc-env.yml

```
@@ -49,37 +49,55 @@ jobs:
         with:
           script: |
             const fs = require('fs');
-            const buildData = fs.readFileSync('master_build_context.json', 'utf8');
+            const buildData = fs.readFileSync('deployment_build_context.json', 'utf8');
             return JSON.parse(buildData);
 
       - name: Parse Push Event Information
         id: pr_info
         if: github.event.workflow_run.event == 'push' && steps.build_context.outputs.result != ''
         run: |
           echo '${{ steps.read_build_info.outputs.result }}' | jq -r '.ref' > merge_ref.txt
+          echo '${{ steps.read_build_info.outputs.result }}' | jq -r '.ref_name' > merge_ref_name.txt
           echo '${{ steps.read_build_info.outputs.result }}' | jq -c '.updated_demos' > updated_demos.json
           echo '${{ steps.read_build_info.outputs.result }}' | jq -c '.deleted_demos' > deleted_demos.json
 
           echo "merge_ref=$(cat merge_ref.txt)" >> $GITHUB_OUTPUT
+          echo "merge_ref_name=$(cat merge_ref_name.txt)" >> $GITHUB_OUTPUT
           echo "updated_demos=$(cat updated_demos.json)" >> $GITHUB_OUTPUT
           echo "deleted_demos=$(cat deleted_demos.json)" >> $GITHUB_OUTPUT
+
+      - name: Determine SWC environment via merge_ref_name
+        if: steps.pr_info.outputs.merge_ref_name != ''
+        id: determine_env
+        run: |
+          if [[ -z "${{ steps.pr_info.outputs.merge_ref_name }}" ]]; then
+            echo "merge_ref_name is not set!"
+            exit 1
+          elif [[ "${{ steps.pr_info.outputs.merge_ref_name }}" == "master" || "${{ steps.pr_info.outputs.merge_ref_name }}" == "test-v2-master" ]]; then
+            echo "swc-env=swc-staging" >> $GITHUB_OUTPUT
+          else
+            echo "swc-env=swc-dev" >> $GITHUB_OUTPUT
+          fi
```

Since we're still testing, maybe a good idea to do:
```suggestion
      - name: Determine SWC environment via merge_ref_name
        if: steps.pr_info.outputs.merge_ref_name != ''
        id: determine_env
        run: |
          # Remove testing branches for production release
          if [[ -z "${{ steps.pr_info.outputs.merge_ref_name }}" ]]; then
            echo "merge_ref_name is not set!"
            exit 1
          elif [[ "${{ steps.pr_info.outputs.merge_ref_name }}" == "master" || "${{ steps.pr_info.outputs.merge_ref_name }}" == "test-v2-master" ]]; then
            echo "swc-env=swc-staging" >> $GITHUB_OUTPUT
          elif [[ "${{ steps.pr_info.outputs.merge_ref_name }}" == "dev" || "${{ steps.pr_info.outputs.merge_ref_name }}" == "test-v2-dev" ]]; then
            echo "swc-env=swc-dev" >> $GITHUB_OUTPUT
          else
            echo "Unexpected merge_ref_name: ${{ steps.pr_info.outputs.merge_ref_name }}"
            exit 1
          fi
```